lib/publiccloud/basetest.pm

```diff
@@ -31,7 +32,10 @@ sub provider_factory {
 
     $args{provider} //= get_required_var('PUBLIC_CLOUD_PROVIDER');
 
-    if ($args{provider} eq 'EC2') {
+    if (get_var('PUBLIC_CLOUD_INSTANCE_IP')) {
+        $provider = publiccloud::noprovider->new();
```

Comment:
I really doubt we need an empty base module to load `existing.pm`. Can't we just skip it and returning noprovider object here somehow?